compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2996,3 +2872,135 @@ impl Location {
         }
     }
 }
+
+#[derive(Clone, TyEncodable, TyDecodable, Debug, HashStable, TypeFoldable)]
```

Comment:
this file is large enough, let's use the opportunity and move the new code to a submodule